libs/language-server/src/lib/ast/wrappers/cell-range-wrapper.ts

```diff
@@ -138,6 +138,37 @@ export class CellRangeWrapper<N extends CellRangeLiteral = CellRangeLiteral>
     return cellRange.cellFrom !== undefined && cellRange.cellTo !== undefined;
   }
 
+  isOneDimensional(): boolean {
+    return (
+      this.from.columnIndex === this.to.columnIndex ||
+      this.from.rowIndex === this.to.rowIndex
+    );
+  }
+
+  numberOfCells(): number {
+    const numberOfRows = this.getNumberOfRows();
+    const numberOfColumns = this.getNumberOfColumns();
+
+    return numberOfRows * numberOfColumns;
+  }
+
+  private getNumberOfRows(): number {
+    if (this.from.rowIndex === LAST_INDEX && this.to.rowIndex === LAST_INDEX) {
```

Comment:
Do we need this? If this check is true, line 159 returns 1 as well (LAST_INDEX minus itself is 0, plus 1 will be 1). Same for number of columns.

Reply:
Yes: `LAST_INDEX` holds the value `Number.POSITIVE_INFINITY`, so without the check we'd compute `Number.POSITIVE_INFINITY - Number.POSITIVE_INFINITY` which yields `NaN`.